package-diff

```diff
@@ -32,81 +37,166 @@ VERSION_B="$2"
 FILESONLY="${FILESONLY-0}"
 CUTKERNEL="${CUTKERNEL-0}"
 CALCSIZE="${CALCSIZE-0}"
+WITHNEWSTUFF="${WITHNEWSTUFF-0}"
+WTD=0
+NEWCALCSIZE=0
 
 A="$(mktemp "/tmp/$VERSION_A-XXXXXX")"
 B="$(mktemp "/tmp/$VERSION_B-XXXXXX")"
 
 trap "rm -f \"$A\" \"$B\"" EXIT
 
-if [ "$FROM_A" = "release" ]; then
-  URL_A="https://${CHANNEL_A}.release.flatcar-linux.net/${BOARD_A}/${VERSION_A}/${FILE}"
-elif [ "$FROM_A" = "bincache" ]; then
-  URL_A="https://bincache.flatcar-linux.net/images/${BOARD_A/-usr/}/${VERSION_A}/${FILE}"
-elif echo "$FROM_A" | grep -q '^file'; then
-  URL_A="${FROM_A}/${VERSION_A}/${FILE}"
-else
-  URL_A="https://bucket.release.flatcar-linux.net/flatcar-jenkins${MODE_A}${CHANNEL_A}/boards/${BOARD_A}/${VERSION_A}/${FILE}"
+function download {
+    local file="${1}"; shift
+
+    if [ "$FROM_A" = "release" ]; then
+        URL_A="https://${CHANNEL_A}.release.flatcar-linux.net/${BOARD_A}/${VERSION_A}/${file}"
+    elif [ "$FROM_A" = "bincache" ]; then
+        URL_A="https://bincache.flatcar-linux.net/images/${BOARD_A/-usr/}/${VERSION_A}/${file}"
+    elif echo "$FROM_A" | grep -q '^file'; then
+        URL_A="${FROM_A}/${VERSION_A}/${file}"
+    else
+        URL_A="https://bucket.release.flatcar-linux.net/flatcar-jenkins${MODE_A}${CHANNEL_A}/boards/${BOARD_A}/${VERSION_A}/${file}"
+    fi
+    if [ "$FROM_B" = "release" ]; then
+        URL_B="https://${CHANNEL_B}.release.flatcar-linux.net/${BOARD_B}/${VERSION_B}/${file}"
+    elif [ "$FROM_B" = "bincache" ]; then
+        URL_B="https://bincache.flatcar-linux.net/images/${BOARD_B/-usr/}/${VERSION_B}/${file}"
+    elif echo "$FROM_B" | grep -q '^file'; then
+        URL_B="${FROM_B}/${VERSION_B}/${file}"
+    else
+        URL_B="https://bucket.release.flatcar-linux.net/flatcar-jenkins${MODE_B}${CHANNEL_B}/boards/${BOARD_B}/${VERSION_B}/${file}"
+    fi
+
+    curl --location --silent -S -o "$A" "$URL_A"
+    curl --location --silent -S -o "$B" "$URL_B"
+
+    if [[ ! -s "${A}" ]] || [[ ! -s "${B}" ]]; then
+        return 1
+    fi
+    return 0
+}
+
+if [[ "${WITHNEWSTUFF}" = '1' ]]; then
```

Comment:
Could we make this the default behavior? With the fallback in place it's ok, or?